localstack/services/logs/provider.py

```diff
@@ -341,20 +348,30 @@ def moto_put_log_events(self: "MotoLogStream", log_events):
             payload_gz_encoded = output.getvalue()
             event = {"awslogs": {"data": base64.b64encode(output.getvalue()).decode("utf-8")}}
 
+            arn_data = parse_arn(destination_arn)
+
             if ":lambda:" in destination_arn:
-                client = connect_to(region_name=extract_region_from_arn(destination_arn)).lambda_
+                client = connect_to(
+                    aws_access_key_id=arn_data["account"], region_name=arn_data["region"]
+                ).lambda_
                 lambda_name = arns.lambda_function_name(destination_arn)
                 client.invoke(FunctionName=lambda_name, Payload=json.dumps(event))
+
             if ":kinesis:" in destination_arn:
-                client = connect_to().kinesis
+                client = connect_to(
+                    aws_access_key_id=arn_data["account"], region_name=arn_data["region"]
+                ).kinesis
                 stream_name = arns.kinesis_stream_name(destination_arn)
                 client.put_record(
                     StreamName=stream_name,
                     Data=payload_gz_encoded,
                     PartitionKey=self.log_group.name,
                 )
```

Comment:
For `PutRecord`, I believe it takes a `StreamARN` parameter, but I don't think it works with `kinesis-mock` and the one server for each account and the call to `get_aws_account_id`. Something to think about for the future and IAM maybe at some point.